owner/manager.go

```diff
@@ -277,11 +288,20 @@ func init() {
 	}
 }
 
-func isContextDone(ctx goctx.Context) bool {
+func contextDone(ctx goctx.Context, err error) error {
```

Comment:
prefer let the contextDone figure out whether context is done. and add a function like ErrorOccured to check the error, one function is better to do just one thing.

Reply:
This function checks whether the context is done， and we treat these error as the context is done. So I put these logical in one function.